src/main/scala/sigmastate/trees.scala

```diff
@@ -404,3 +404,16 @@ case class If[T <: SType](condition: Value[SBoolean.type], trueBranch: Value[T],
   override lazy val second = trueBranch
   override lazy val third  = falseBranch
 }
+
+
+case class Min[T <: SNumericType](left: Value[T], right: Value[T])
```

Comment:
@ergomorphic Any objections to making `min`, `max` nodes case classes? I've ruled out using `ArithOp`, too much difference. Maybe `ComparisonSelector` parametrized by an op code?

Reply:
@greenhat looks like ArithOp to me, what problem you see?

Reply:
@ergomorphic I cannot remember now, probably not that significant of a difference. I've been trying to pull it through the `Apply(Ident...` which felt very unnatural (parametrization with concrete numeric type, etc.). I'll try `ArithOp` now. Thanks.